### PR TITLE
Embed `SpiServerCore` into tasks with exclusive peripheral ownership

### PR DESCRIPTION
**Pull request by mkeeter**
This PR allows for tasks to exclusively own their SPI peripheral, instead of talking to a separate SPI server task.

It includes code changes to the following 4 tasks:
- `drv-stm32h7-sprot-server`
-  `task-monorail-server`
-  `drv-fpga-server`
-  `task-net`

There exist other tasks which use SPI but don't have exclusive use of a peripheral, e.g. `gimlet-seq-server`; such tasks are left unchanged for now.

In general, owning the SPI peripheral is enabled with the `use-spi-core` feature, which also requires wiring up the SPI IRQ, adding it to the `uses = [...]` list, and adding `config.spi.global_config = "spiX"` to the task.

In addition to the code changes, this PR updates a bunch of `app.toml` files to eliminate separate SPI servers whenever possible.

I'd appreciate someone (@andrewjstone ?) testing this branch on an actual SP + RoT, since I'm not as familiar with that system.

### CODE REVIEW

**Review by jgallagher — COMMENTED** [review]
Code changes all look good to me. I'm holding off on approving so @andrewjstone can give this a whirl with his RoT update branch.

**Review by andrewjstone — APPROVED** [review]
Seems to work great with SpRot!

**Pull request merged**
